toolkit/pallets/block-production-log/src/lib.rs

```diff
@@ -0,0 +1,94 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+pub mod mock;
+#[cfg(test)]
+pub mod test;
+
+mod benchmarking;
+
+pub mod weights;
+
+pub use pallet::*;
+pub use weights::WeightInfo;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use sp_block_production_log::*;
+	use sp_consensus_slots::Slot;
+	use sp_runtime::BoundedVec;
+	use sp_std::vec::Vec;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		#[pallet::constant]
+		type MaxLogSize: Get<u32>;
+
+		#[pallet::constant]
+		type MaxProducerIdLength: Get<u32>;
+
+		type BlockProducerId: Member
+			+ Parameter
+			+ MaxEncodedLen
+			+ From<BoundedVec<u8, Self::MaxProducerIdLength>>;
```

Comment:
I admit I wish to not have this one, added only because without a `From` I had a problem creating instance in benchmarks/tests.

Reply:
It is possible to add additional parameterizations in benchmarks, maybe it will work:
https://paritytech.github.io/polkadot-sdk/master/frame_benchmarking/v2/index.html#where-clause

Reply:
Works like a charm, thank you!